vault.py

```diff
@@ -72,15 +73,18 @@ def run(self, terms, inject=None, variables=None, **kwargs):
         # so as not to encourage bad security practices.
         token = os.getenv('VAULT_TOKEN')
         if not token:
+            token_path = os.path.join(os.getenv('HOME'), '.vault-token')
             try:
-                with open(os.path.join(os.getenv('HOME'), '.vault-token')) as file:
-                    token = file.read().strip()
-            except IOError:
-                # token not found in file is same case below as not found in env var
-                pass
+                with open(token_path) as token_file:
+                    token = token_file.read().strip()
+            except IOError as err:
+                if err.errno == errno.ENOENT:
```

Comment:
I'm not sure what you're checking with this if statement. It will do nothing if the condition is met and continue to raise the exception below in any case. That means the `if not token` statement on line 84 wont be reached.